tests/5.1/metadirective/test_metadirective_target_device.c

```diff
@@ -0,0 +1,63 @@
+//===--------------------- test_metadirective_target_device.c ---------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+// 
+// Tests that target_device is recognized & will work properly when matching an nvidia
+// GPU.
+//
+////===---------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int test_metadirective_target_device() {
+   int errors = 0;
+   int A[N];
+
+   for (int i = 0; i < N; i++) {
+      A[i] = 1;
+   }
+
+   #pragma omp target map(tofrom: A)
+   {
+      #pragma omp parallel num_threads(4)
+      {
+      // Except that one of these are true, so the array should be set to masked thread num (0)
+      #pragma omp metadirective \
+         when( target_device={kind(nohost)}: masked ) \
```

Comment:
should be
device={arch("nvptx")...

Reply:
This is a test for target_device, if I am understanding your comment correctly